src/handlers/records-read.ts

```diff
@@ -37,13 +37,9 @@ export class RecordsReadHandler implements MethodHandler {
       return messageReplyFromError(e, 401);
     }
 
-    // get existing messages matching `recordId` so we can perform authorization
-    const query = {
-      interface : DwnInterfaceName.Records,
-      recordId  : message.descriptor.recordId
-    };
+    // get existing messages matching `recordId` or `protocol` with `protocolPath` so we can perform authorization
+    const query: Filter = RecordsReadHandler.createFilter(recordsRead.message.descriptor);
     const existingMessages = await this.messageStore.query(tenant, query) as TimestampedMessage[];
```

Comment:
With this PR, the returned messages could be in 100s or even 1000s, and is subject to sort/paging. What were the scenario that prompted this change again? If it's for path that will ever have only one record, should we add a check for it?

Reply:
The scenario is just to get the latest record under the path, but it doesn't have to be a "$singleton" record, we are moving away from singleton and toward $keep instead, which could have a value greater than 1.

So I don't think there are any checks here to provide, however we would add a limit to 1 and sort appropriately when that is exposed.

Reply:
> The scenario is just to get the latest record under the path

This sounds more like a "requirement" but not scenario to me, do you know what the/an actual scenario that called for this requirement was?